qcengine/programs/molpro.py

```diff
@@ -201,49 +224,43 @@ def parse_output(self, outfiles: Dict[str, str], input_model: 'ResultInput') ->
-            "triplet pair energy": "mp2_triplet_pair_energy"
-        }
-
-        # CCSD maps
-        ccsd_energy_map = {"total energy": "ccsd_total_energy", "correlation energy": "ccsd_correlation_energy"}
-        ccsd_dipole_map = {"Dipole moment": "ccsd_dipole_moment"}
-        ccsd_extras = {
-            "singlet pair energy": "ccsd_singlet_pair_energy",
-            "triplet pair energy": "ccsd_triplet_pair_energy"
-        }
-
-        # CCSD(T) maps
-        # TODO There are two instances of "correlation energy" in the xml. One for CCSD and one for CCSD(T)
-        #      Will need a bit more logic to separate between the two
-        ccsd_prt_pr_energy_map = {
-            "total energy": "ccsd_prt_pr_total_energy",
-            "correlation energy": "ccsd_prt_pr_correlation_energy",
-        }
-        ccsd_prt_pr_dipole_map = {"Dipole moment": "ccsd_prt_pr_dipole_moment"}
-        ccsd_prt_pr_extras = {**ccsd_extras, "contribution": "prt_pr_contribution"}
-
-        # Compiling the method maps
-        scf_maps = {"energy": scf_energy_map, "dipole": scf_dipole_map, "extras": scf_extras}
-        mp2_maps = {"energy": mp2_energy_map, "dipole": mp2_dipole_map, "extras": mp2_extras}
-        ccsd_maps = {"energy": ccsd_energy_map, "dipole": ccsd_dipole_map, "extras": ccsd_extras}
-        ccsd_prt_pr_maps = {
-            "energy": ccsd_prt_pr_energy_map,
-            "dipole": ccsd_prt_pr_dipole_map,
-            "extras": ccsd_prt_pr_extras
+        # Molpro map
+        molpro_map = {
+            "Energy": {
+                "HF": "scf_total_energy",
+                "RHF": "scf_total_energy",
+                "KS": "scf_total_energy",
+                "RKS": "scf_total_energy"
+            },
+            "total energy": {
+                "MP2": "mp2_total_energy",
+                "CCSD": "ccsd_total_energy",
+                "CCSD(T)": "ccsd_prt_pr_total_energy"
+            },
+            "correlation energy": {
+                "MP2": "mp2_correlation_energy",
+                "CCSD": "ccsd_correlation_energy",
+                "CCSD(T)": "ccsd_prt_pr_correlation_energy",  # Need both CCSD(T) and Total
+                "Total": "ccsd_prt_pr_correlation_energy"  # Total corresponds to CCSD(T) correlation energy
+            },
+            "singlet pair energy": {
+                "MP2": "mp2_singlet_pair_energy",
+                "CCSD": "ccsd_singlet_pair_energy"
+            },
+            "triplet pair energy": {
+                "MP2": "mp2_triplet_pair_energy",
+                "CCSD": "ccsd_triplet_pair_energy"
+            },
+            "Dipole moment": {
+                "HF": "scf_dipole_moment",
+                "RHF": "scf_dipole_moment",
+                "KS": "scf_dipole_moment",
+                "RKS": "scf_dipole_moment",
+                "MP2": "mp2_dipole_moment",
+                "CCSD": "ccsd_dipole_moment",
+                "CCSD(T)": "ccsd_prt_pr_dipole_moment"
+            }
```

Comment:
+1 for making this a literal.

I probably asked this before, but are the SCF energy components available in the MOLPRO XML?

Reply:
Actually yes they are I just noticed. At least for the DFT case the XC functional is available, but in a different part of the xml from where I grab the total energy. I can see if I can grab these components as well.

Reply:
So I was able to grab `nuclear_repulsion_energy` and the `scf_xc_energy`, but that's all I can get for the scf energy breakdown.

Reply:
Thanks, always worth pulling anything else that we can!